src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package taxi.service;
+
+import java.util.Optional;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driverOptional = driverService.findByLogin(login);
+        Driver driver = driverOptional.orElseThrow(() -> {
+            return new AuthenticationException("Login or password is incorrect");
+        });
+        if (driver.getPassword().equals(password)) {
+            return driver;
+        }
+        throw new AuthenticationException("Login or password is incorrect");
```

Comment:
This is code duplication. Avoid it using one IF block (orElseThrow is hidden IF in fakt)
![image](https://user-images.githubusercontent.com/20326858/216301816-c06abc99-b0dd-447c-875c-1b28b5cc85a3.png)